### PR TITLE
aws_elasticloadbalancingv2_listener_rule

### PR DESCRIPTION
**Pull request by soumyo13**
### Description

It contains all the singular and plural resources of the above resource.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_elasticloadbalancingv2_listener_rule.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+## Syntax
+
+Ensure that a rule exists.
+    describe aws_elasticloadbalancingv2_listener_rule(rule_arns: 'test_rule') do
+      it { should exist }
+    end
+
+## Parameters
+
+`rule_arns` _(required)_
```
Comment (IanMadd) [inline docs/resources/aws_elasticloadbalancingv2_listener_rule.md:21]: Does this take one arn or multiple arns?

File: docs/resources/aws_elasticloadbalancingv2_listener_rules.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+## Syntax
+
+Ensure that a listener_arn exists.
+    describe aws_elasticloadbalancingv2_listener_rules(listener_arn: 'test1') do
+      it { should exist }
+    end
+
+## Parameters
+
```
Comment (IanMadd) [inline docs/resources/aws_elasticloadbalancingv2_listener_rules.md:20]: Does this require the listener arn?

Comment (Nirbhay1997) [reply docs/resources/aws_elasticloadbalancingv2_listener_rules.md:21]: yes updated

File: docs/resources/aws_elasticloadbalancingv2_listener_rule.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+| actions (redirect_config (query)) | The query parameters, URL-encoded when necessary, but not percent-encoded. Do not include the leading "?", as it is automatically added. You can specify any of the reserved keywords. |
+| actions (redirect_config (status_code)) | The HTTP redirect code. The redirect is either permanent (HTTP 301) or temporary (HTTP 302). |
+| actions (fixed_response_config (message_body)) | The message from a custom HTTP response. Only valid when Type is `fixed-response`. |
+| actions (fixed_response_config (status_code)) | The HTTP response code (2XX, 4XX, or 5XX) from a custom HTTP response. |
+| actions (fixed_response_config (content_type)) | The content type from a custom HTTP response. Valid values: `text/plain`, `text/css`, `text/html`, `application/javascript`, `application/json`. |
+| actions (fixed_response_config (target_groups (target_group_arn))) | The ARN of the target group. |
+| actions (fixed_response_config (target_groups (weight))) | The weight of the target group. The range is 0 to 999. |
```
Comment (IanMadd) [inline docs/resources/aws_elasticloadbalancingv2_listener_rule.md:75]:
```suggestion
| actions (forward_config (target_groups ( target_group_tupple (target_group_arn)))) | The ARN of the target group. |
| actions (forward_config (target_groups ( target_group_tupple (weight)))) | The weight of the target group. The range is 0 to 999. |
```

File: docs/resources/aws_elasticloadbalancingv2_listener_rules.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+| conditions   | The conditions for listener rules. Each rule can include zero or one of the following conditions: `http-request-method` , `host-header` , `path-pattern` , and `source-ip`, and zero or more of the following conditions: `http-header` and `query-string`. |
+| actions      | The actions for listener rules. Each rule includes exactly one of the following types of actions: `forward`, `redirect`, or `fixed-response`. |
+| is_defaults  | Indicates whether this is the default rule. |
+
+## Examples
+
+### Ensure a listener ARN is available.
+
+    describe aws_elasticloadbalancingv2_listener_rules(listener_arn: 'LISTENER_ARN') do
+      its('names') { should include 'test1' }
```
Comment (IanMadd) [inline docs/resources/aws_elasticloadbalancingv2_listener_rules.md:41]: Is names valid?

**Review by IanMadd — APPROVED** [review]

File: docs/resources/aws_elasticloadbalancingv2_listener_rules.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+`listener_arn`  _(required)_
+
+For additional information, see the [AWS documentation on ELBv2 Listener Rule](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listenerrule.html).
+
+## Properties
+
+| Property | Description|
+| --- | ---    |
+| listener_arn | The ARN of the listener. |
```
Comment (sathish-progress) [inline docs/resources/aws_elasticloadbalancingv2_listener_rules.md:30]: please add fields

**Review by sathish-progress — APPROVED** [review]
LGTM, please update readme

File: docs/resources/aws_elasticloadbalancingv2_listener_rules.md (hunk 4 above)
Comment (soumyo13) [reply docs/resources/aws_elasticloadbalancingv2_listener_rules.md]: Corrected to 'rule_arns'

File: docs/resources/aws_elasticloadbalancingv2_listener_rules.md (hunk 5 above)
Comment (soumyo13) [reply docs/resources/aws_elasticloadbalancingv2_listener_rules.md]: Done

**Pull request merged**